MAD-X/PALS Load File: `min_model=...` (#1593)
BLAST-ImpactX/impactx: docs/source/usage/python.rst

      Add a single element to the list.

   .. py:method:: load_file(filename, nslice=1, *, min_model="linear")

      Load and append a lattice file from MAD-X (.madx) or PALS (e.g., .pals.yaml) formats.

      The reader picks the *cheapest* ImpactX element that represents each imported
      element.
      ``min_model`` raises the lower gate of that choice, so that an imported lattice
      starts out with at least the requested level of physical fidelity
      (see :ref:`theory-assumptions`):

>       * ``"linear"`` (default) reproduces the historic behavior

Author: ```suggestion
      * ``"linear"`` (default) uses the linear model
```

      * ``"paraxial"`` requires at least the chromatic ``Chr*`` models
      * ``"exact"`` requires at least the exact-Hamiltonian ``Exact*`` models

      A floor never *lowers* a model.

>       An element that already requires a richer model, such as a skew quadrupole

Reviewer: Oh, this statement is false.  A skew quad can be represented as a linear Quad, paraxial ChrQuad, or ExactQuad with a 45 deg rotation (`tilt`).

Author: thx! fixed

>       An element that already requires a richer model, such as a skew quadrupole

Reviewer: ```suggestion
      An element that already requires a richer model, such as a thick octupole
```

Author: thx! fixed

           - ``Drift``
           - ``ChrDrift``
           - ``ExactDrift``
         * - ``QUADRUPOLE``
           - ``Quad``
           - ``ChrQuad``
           - ``ExactQuad``
         * - ``SBEND``, ``RBEND``
           - ``Sbend`` / ``CFbend``
           - ``ExactSbend`` / ``ExactCFbend``
           - ``ExactSbend`` / ``ExactCFbend``
         * - ``DIPEDGE`` (and bend edges)
           - ``DipEdge(model="linear")``
           - ``DipEdge(model="nonlinear")``
           - ``DipEdge(model="nonlinear")``
         * - ``SOLENOID``
           - ``Sol``
           - ``Sol`` (warns)
           - ``Sol`` (warns)
         * - ``SEXTUPOLE``, ``OCTUPOLE``, skew ``QUADRUPOLE``
           - ``ExactMultipole``
           - ``ExactMultipole``
           - ``ExactMultipole``

      Elements that MAD-X itself defines as thin (``MULTIPOLE``, ``KICKER``,
      ``RFCAVITY``, ``NLLENS``) and structural elements (``MARKER``, apertures,
      monitors) have no model tiers, so the kick or map they translate into is
      the same at every floor.
      Their *length*, however, is carried by drifts that the reader adds around
      that kick, and those follow ``min_model`` like any other drift.

>       A finite-length ``RFCAVITY``, for example, becomes

Author: ```suggestion
      A finite-length ``RFCAVITY`` is a thin kick model, for example, becomes
```

      ``ExactDrift`` + ``ShortRF`` + ``ExactDrift`` at ``min_model="exact"``.
      The same applies to the drifts emitted for ``COLLIMATOR``, ``INSTRUMENT``,

>       ``PLACEHOLDER`` and for a ``MONITOR`` with ``L > 0``.

Author: To check with @cemitch99 if that is a sensible element tier promotion path.
